### ISSUE TITLE
Move this repo to https://github.com/jakartaee

### ISSUE THREAD

**Issue by arjantijms** [jakartaee/cdi#610]
In the steering committee we voted to move all specification projects from eclipse-ee4j to jakartaee. As such this repo should move to in due time.

Moving is fairly automatic, and redirects will be automatically put in place from the current location here to the new location.

If someone can authorise the move (e.g. @manovotn & @Ladicek) I can take care of it if needed.

See also:

"We voted on it in 2019 and then in 2020 we added it explicitly to our 2021 plan and voted on that plan.  I think we just need to start executing.  I sent about 30 emails like the following in an attempt to add some shape to it and kick off the move in the spec projects. 

 - https://www.eclipse.org/lists/jakartaee-platform-dev/msg02940.html
 - https://www.eclipse.org/lists/messaging-dev/msg00028.html
 - https://www.eclipse.org/lists/jakartaee-tck-dev/msg01417.html
 - https://www.eclipse.org/lists/ejb-dev/msg00264.html
"

From: https://www.eclipse.org/lists/jakarta.ee-spec/msg02073.html

Also note that many projects have indeed moved: https://github.com/jakartaee/

**Comment by manovotn** [comment]
> Moving is fairly automatic, and redirects will be automatically put in place from the current location here to the new location.

So the project permissions will be kept as well? I.e. committer status. Or do we need to take some extra step there?

> If someone can authorise the move (e.g. @manovotn & @Ladicek) I can take care of it if needed.

I am not sure what kind of authorization do you need, if it is just an 'ack' then by all means - if the committee passed the vote then we indeed need to move.
If you need any formal authorization, than that should probably come from project lead which is @starksm64

**Comment by arjantijms** [comment]
An "ack" from a committer is fine. It has already been formally authorised essentially, but it's up to the project teams (and therefor the individual committers) to decide when to exactly move and start that process.

**Comment by arjantijms** [comment]
> So the project permissions will be kept as well? I.e. committer status. Or do we need to take some extra step there?

Yes, those will all be kept. The only thing we might want to do (should, actually), is update the CI at http://ci.eclipse.org/cdi to use the new git repo address. Most tools are fine with being redirected, and will continu to function, but occasionally the CI doesn't like the redirect and chokes on it. So to prevent any issues it's best to update that one.

**Comment by arjantijms** [comment]
Repo has been moved to https://github.com/jakartaee/cdi.

Committers will (re-)gain access once the sync script runs (~2h).

To be 100% sure,  update all local and CI configurations to point to the new URL.

**Comment by Ladicek** [comment]
We should also move https://github.com/eclipse-ee4j/cdi-tck/ I guess?

**Comment by arjantijms** [comment]
> We should also move https://github.com/eclipse-ee4j/cdi-tck/ I guess?

I was just looking at that ;) Indeed, I overlooked it above since almost all other projects have TCK as a folder in the spec project.

What do you think, for consistency reasons, does it make sense for CDI to have a /tck folder too?

See e.g. 

* https://github.com/jakartaee/faces/tree/master/tck
* https://github.com/jakartaee/security/tree/master/tck
* https://github.com/jakartaee/authentication/tree/master/tck

**Comment by Ladicek** [comment]
I don't really mind. Since the CDI TCK is a rather big project with its own API and SPI and everything, I guess I'd rather keep it separate, though merging them would have some obvious advantages...

**Comment by Emily-Jiang** [comment]
I prefer to have a TCK folder under cdi to keep it consistent with other specs (having their tcks out of the platform tck) such as jax-rs etc. In this case, it is easier to navigate.

**Comment by arjantijms** [comment]
It's a big project especially with all the ArchiveBuilder etc utility code, but the API and SPIs are really small.

Slightly off-topic, but since it came up here; as the SPI needs to be implemented by definition by every CDI implementation anyway, could it not be moved to the main CDI spec? Something like `org.jboss.cdi.tck.spi.Beans#isProxy` would probably be welcomed by more advanced frameworks anyway (if I remember well, there even was an ask for this).

**Comment by arjantijms** [comment]
>  Note that it holds not only CDI but also interceptors TCKs.

But the repo is called `cdi-tck`. 

Additionally, in-time, maybe we want to consider moving interceptors to be a sub-project of CDI?

**Comment by manovotn** [comment]
> > Note that it holds not only CDI but also interceptors TCKs.
> 
> But the repo is called `cdi-tck`.
> 
> Additionally, in-time, maybe we want to consider moving interceptors to be a sub-project of CDI?

I don't know the origins of the repo name but I presume interceptors TCKs landed here because there if no standalone implementations and you will end up running with CDI anyway.

As for moving interceptors - isn't EJB using them as well?

> p.s. @manovotn if separate repos for TCKs are better, does that mean we should also create separate repos for the Faces and Security TCKs? From the look of it the Faces TCK is as big, if not bigger, than the CDI TCK.

Why should we? To my knowledge this is governed by each project and we can keep it that way. Different projects, different preferences. It is not like it's hard to locate the TCK for any of the projects :)

**Comment by arjantijms** [comment]
> As for moving interceptors - isn't EJB using them as well?

They are, and it was originally a sub-project of EJB. But EJB is on a trajectory of being made a stable API (e.g. being phased out). 

> Why should we?

For consistency?

> Different projects, different preferences.

I understand, but I'm just wondering then why CDI needs its TCK to be separate, but Faces does not.

My guess is that it's not even an explicit preference. Someone just did it at some point (e.g. it's just historic). I guess if CDI has just happened to have it as a folder, and Faces just happened to have it as a separate repo, the preference was to keep CDI a folder?

In other words, it's perhaps less about a preference, and more about conservatism?

**Comment by manovotn** [comment]
> They are, and it was originally a sub-project of EJB. But EJB is on a trajectory of being made a stable API (e.g. being phased out).

Once EJB is a goner, we might do that but up until then it is needed by both APIs.

> For consistency?

Like I said, it isn't hard to find a TCK belonging to a certain project.
There is no actual harm done if projects do it slightly differently.

> I understand, but I'm just wondering then why CDI needs its TCK to be separate, but Faces does not.

The *need* and *preference* are not the same. Neither of them *needs* to be a folder or a project.
I just find a separate project to be a cleaner separation between the the two. Both repos have slightly different focuses and issues. With CDI repo you will issue feature proposals, design questions, doc adjustments etc/ It is where a user might go to propose something.
Whereas TCK is where you go to challenge existing tests or the scope of their coverage; a kind of issue that standard user won't care about much but implementors and integrators will.

Another reason is that the release cycle can be separate and transparent. We can and do issue SP releases of TCK based on challenges even now and CDI API doesn't need to be touched.

> In other words, it's perhaps less about a preference, and more about conservatism?

Not really, conservatism would be if I opposed the move under different GH org even though there are no actual differences in the end ;-)

**Comment by arjantijms** [comment]
> Once EJB is a goner, we might do that but up until then it is needed by both APIs.

I might be wrong here, but I think it's possible for EJB to depend on a sub-spec.

> Whereas TCK is where you go to challenge existing tests or the scope of their coverage;
> Another reason is that the release cycle can be separate and transparent. 

I understand all that, but what I don't understand is why that would only hold for CDI? Don't Faces, REST, etc need their release cycle to be separate and transparent?

Why does CDI, as one of the only projects, need a separate repo, where (seemingly) the other projects don't need it?

I guess REST is as active and much used as CDI, whereas Faces is at least as big as CDI, so neither usage nor size seem to be factors here. Please help me understand what makes CDI so different.

**Comment by manovotn** [comment]
> Please help me understand what makes CDI so different.

Preferences of its project contributors? There doesn't seem to be a given rule of how jakarta wants to structure projects, so please help me understand what's so difficult about doing it in a way that's agreed on in any given project by its committers/contributors.

**Comment by arjantijms** [comment]
> Preferences of its project contributors? 

Isn't @Emily-Jiang a project contributor? Have we asked @struberg about this? What does @graemerocher think? Or @ljnelson ? Maybe @antoinesd has something to say?

I've only heard @manovotn & @Ladicek here. I mean, this is not the end of the world, and if the committers really want a separate repo, then sure, who am I to even suggest going against that. But I was just wondering whether this is really an explicit wish, or more like a pavlov's response "No, corner case, won't do"?

**Comment by ljnelson** [comment]
I've always wondered why the tck is not in this repo.  I would prefer it to be in this repo.

**Comment by pzygielo** [comment]
> moving the cdi-tck repo now into a folder inside the cdi repo is not trivial as we want to keep the history

I suppose `git merge --allow-unrelated-histories` could help with that.

**Comment by manovotn** [comment]
> @manovotn it seems like you haven't accepted the invitation for the new org/repo yet. That should be the reason why you don't have access now

Thanks, that email slipped my filters somehow; works fine now.

**Comment by manovotn** [comment]
@arjantijms do you think we can close this issue now?

**Comment by arjantijms** [comment]
@manovotn yes indeed, if we want any follow up for merging the TCK repos we can create new issues.
